apps/wallet/src/ui/app/pages/swap/GasFeeSection.tsx

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+import {
+	Coins,
+	getUSDCurrency,
+	SUI_CONVERSION_RATE,
+	useBalanceConversion,
+} from '_app/hooks/useDeepBook';
+import { Text } from '_app/shared/text';
+import { DescriptionItem } from '_pages/approval-request/transaction-request/DescriptionList';
+import { GAS_TYPE_ARG } from '_redux/slices/sui-objects/Coin';
+import { useCoinMetadata, useFormatCoin } from '@mysten/core';
+import { SUI_TYPE_ARG } from '@mysten/sui.js/utils';
+import BigNumber from 'bignumber.js';
+import { useMemo } from 'react';
+
+import { FEES_PERCENTAGE } from './utils';
+
+export function GasFeeSection({
+	activeCoinType,
+	totalGas,
+	amount,
+	isValid,
+}: {
+	activeCoinType: string | null;
+	amount: string;
+	isValid: boolean;
+	totalGas: string;
+}) {
+	const { data: activeCoinData } = useCoinMetadata(activeCoinType);
+	const isAsk = activeCoinType === SUI_TYPE_ARG;
+
+	const estimatedFees = useMemo(() => {
+		if (!amount || !isValid) {
+			return null;
+		}
+
+		return new BigNumber(amount).times(FEES_PERCENTAGE);
+	}, [amount, isValid]);
+
+	const { rawValue } = useBalanceConversion(
+		estimatedFees,
+		isAsk ? Coins.SUI : Coins.USDC,
+		isAsk ? Coins.USDC : Coins.SUI,
+		isAsk ? -SUI_CONVERSION_RATE : SUI_CONVERSION_RATE,
+	);
+
+	const [gas, symbol] = useFormatCoin(totalGas, GAS_TYPE_ARG);
+
+	const formattedEstimatedFees = getUSDCurrency(rawValue);
+
+	return (
+		<div className="flex flex-col border border-hero-darkest/20 rounded-xl p-5 gap-4 border-solid">
+			<DescriptionItem
+				title={
+					<Text variant="bodySmall" weight="medium" color="steel-dark">
+						Fees ({FEES_PERCENTAGE}%)
+					</Text>
+				}
+			>
+				<Text variant="bodySmall" weight="medium" color="steel-darker">
+					{estimatedFees
+						? `${estimatedFees.toLocaleString()} ${activeCoinData?.symbol} (${formattedEstimatedFees})`
+						: '--'}
+				</Text>
+			</DescriptionItem>
+
+			<div className="bg-gray-40 h-px w-full" />
+
+			<DescriptionItem
+				title={
+					<Text variant="bodySmall" weight="medium" color="steel-dark">
+						Estimated Gas Fee
+					</Text>
+				}
+			>
+				<Text variant="bodySmall" weight="medium" color="steel-darker">
+					{totalGas && isValid ? `${gas} ${symbol}` : '--'}
```

Comment:
<img width="378" alt="image" src="https://github.com/MystenLabs/sui/assets/122397493/c1539c0e-44e9-46bc-b37a-e5646341ffd7">
gas fees need formatting

Reply:
@mamos-mysten should be fixed in the newest build

<img width="400" alt="Screenshot 2023-10-18 at 3 33 48 PM" src="https://github.com/MystenLabs/sui/assets/127577476/f20f4b9e-9c7f-48fb-895d-d3490c44bcfa">